src/components/LocationPage/ChartsHolder.tsx

```diff
@@ -164,155 +196,14 @@ const ChartsHolder = (props: {
-              <ChartLocationName>{projection.locationName}</ChartLocationName>
-              <ChartDescription>
-                {positiveTestsStatusText(projection)}
-              </ChartDescription>
-              {isMobile && testPositiveData && (
-                <ShareButtons
-                  chartIdentifier={Metric.POSITIVE_TESTS}
-                  {...shareButtonProps}
-                />
-              )}
-              {testPositiveData && (
-                <>
-                  <ChartPositiveTestRate columnData={testPositiveData} />
-                  <Disclaimer metricName={Metric.POSITIVE_TESTS} />
-                </>
-              )}
-              <ChartHeaderWrapper>
-                <ChartHeader ref={icuUtilizationRef}>
-                  {getMetricName(Metric.HOSPITAL_USAGE)}
-                  <BetaTag>Beta</BetaTag>
-                </ChartHeader>
-                {!isMobile && icuUtilizationData && (
-                  <ShareButtons
-                    chartIdentifier={Metric.HOSPITAL_USAGE}
-                    {...shareButtonProps}
-                  />
-                )}
-              </ChartHeaderWrapper>
-              <ChartLocationName>{projection.locationName}</ChartLocationName>
-              <ChartDescription>
-                {hospitalOccupancyStatusText(projection)}
-              </ChartDescription>
-              {isMobile && icuUtilizationData && (
-                <ShareButtons
-                  chartIdentifier={Metric.HOSPITAL_USAGE}
-                  {...shareButtonProps}
-                />
-              )}
-              {icuUtilizationData && (
-                <>
-                  <ChartICUHeadroom columnData={icuUtilizationData} />
-                  <Disclaimer metricName={Metric.HOSPITAL_USAGE} />
-                </>
-              )}
-              <ChartHeaderWrapper>
-                <ChartHeader ref={contactTracingRef}>
-                  {getMetricName(Metric.CONTACT_TRACING)}
-                  <BetaTag>Beta</BetaTag>
-                </ChartHeader>
-                {!isMobile && contactTracingData && (
-                  <ShareButtons
-                    chartIdentifier={Metric.CONTACT_TRACING}
-                    {...shareButtonProps}
-                  />
-                )}
-              </ChartHeaderWrapper>
-              <ChartLocationName>{projection.locationName}</ChartLocationName>
-              <ChartDescription>
-                {contactTracingStatusText(projection)}
-              </ChartDescription>
-              {isMobile && contactTracingData && (
-                <ShareButtons
-                  chartIdentifier={Metric.CONTACT_TRACING}
-                  {...shareButtonProps}
+              {chartPropsForMap.map(chartProps => (
+                <ChartBlock
+                  noInterventionProjection={noInterventionProjection}
+                  projection={projection}
+                  projections={props.projections}
+                  {...chartProps}
```

Comment:
I think (especially if you take my suggestions to limit the properties we need to pass in) then we don't need to pre-generate chartProps for all the metrics.  We can just change this code to something like:

```
ALL_METRICS.map(metric => (
  <ChartBlock
    metric={metric}
    projections={props.projections}
    chartRef={metricRefs[metric]}
    shareButtonProps={shareButtonProps}
    />
  )
```

Note that I invented a new chartRefs map.  You could define it above as:

```
  const metricRefs: { [metric in Metric]: React.RefObject<HTMLDivElement> } = {
    [Metric.CASE_GROWTH_RATE]: rtRangeRef,
    [Metric.POSITIVE_TESTS]: testPositiveRef,
    [Metric.HOSPITAL_USAGE]: icuUtilizationRef,
    [Metric.CONTACT_TRACING]: contactTracingRef,
    [Metric.FUTURE_PROJECTIONS]: futureProjectionsRef
  };
```

Or we could probably even get rid of the individual `fooRef` variables and just have the `metricRefs` map, but that'd be a bigger refactoring.

Reply:
oooo yes. adding TODO